### ISSUE TITLE
Enhance Software Architecture Draft - Detail Machine Learning Pipeline

### ISSUE THREAD

**Issue by dominic0df** [amosproj/amos2024ss08-cloud-native-llm#17]
## User story
1. As development team
2. I need to decide on a software architecture for the Machine Learning Pipeline with which our LLM will be trained
3. So that we achieve a common usage of technologies and our software is more flexible and easier to work on in pieces.

## Acceptance criteria
* Research is done on how to use Named Entity Recognition (NER) to find key entities with which the data can be classifed into predefined categories
* The already existing architecture is more elaborate about how the Machine Learning Pipeline will work
* Rework on architecture draft
  * One software developer (preferably the Software Architect that created the Architecture Draft) presents the architecture draft with the POs to our industry partner (in a meeting with the POs and the industry partner)
  * Possible suggestions of improvement will be discussed with the team and then possibly the architecture documentation will be updated

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
